### DISCUSSION TITLE
Program won't run in vscode while program correctly compiled and works on other IDEs

### DISCUSSION THREAD

**Discussion by mohamedrhoulam** [microsoft/vscode-cpptools#9749] [Q&A]
So I installed vscode and mingw using mysys2 (as being told in: https://code.visualstudio.com/docs/cpp/config-mingw). After checking if my compiler is installed using `g++ --version` and `gdb --version` I get the expected output I am trying to run the same sample program when clicking on the play button I get this blue text with the output
![image](https://user-images.githubusercontent.com/90448633/185472095-3959da8a-b6bf-4a06-bda0-5a7112da055e.png)

my **task.json:**

`{
    "tasks": [
        {
            "type": "cppbuild",
            "label": "C/C++: g++.exe build active file",
            "command": "C:\\msys64\\mingw64\\bin\\g++.exe",
            "args": [
                "-fdiagnostics-color=always",
                "-g",
                "${file}",
                "-o",
                "${fileDirname}\\${fileBasenameNoExtension}.exe"
            ],
            "options": {
                "cwd": "${fileDirname}"
            },
            "problemMatcher": [
                "$gcc"
            ],
            "group": {
                "kind": "build",
                "isDefault": true
            },
        }
    ],
    "version": "2.0.0"
}`
I have been using CLion and never had an issue with this, also note that I had to reinstall my compiler to check if it would solve the problem. but it didn't work on vscode.

**Comment by mohamedrhoulam** [comment]
Some additional info: when I run it for the first time after oppening vscode it shows the debug terminal for whatever reason:
![image](https://user-images.githubusercontent.com/90448633/185473378-acc03dfa-3b17-4b47-8344-183aacf484d6.png)

Here is the output in the terminal:

Also note that when I run `g++ -o helloworld helloworld.cpp` then `.\helloworld` the program compiles and runs with no issues:

![image](https://user-images.githubusercontent.com/90448633/185472958-e58839d2-b8c6-4bbe-ba18-fa44e3f8110d.png)

**Reply by sean-mcmanus** [reply]
The blue text is normal, expected output from the debugger. You can just ignore it. What is the error you're having? It looks your program is running.

**Reply by mohamedrhoulam** [reply]
I guess it's part `'--stderr=Microsoft-MIEngine-Error-hsnav4kh.srl'` that is bothering me. Also I thought I was supposed to get the output only when I am debugging the program and not when I am running it w/out debug.

**Reply by sean-mcmanus** [reply]
@WardenGnaw Do you know about that?

**Reply by WardenGnaw** [reply]
That is how terminal redirection works for VS Code.

The C/C++ Debug Adapter is redirecting all stdin, stdout, and stderr to named pipes. 

It is appearing on the first time you run the debugger, if you clear the terminal and run it in the terminal without debugging you should not see it.

**Reply by mohamedrhoulam** [reply]
It still appears when I run without debug multiple times consecutively, but anyways I will try a clean uninstall of vscode and my compiler to see if it will fix things correctly, thank you for your help :))
edit: i'll leave this thread unanswered just in case someone who also came across the same has found a solution

**Reply by DhruvgiriGoswami** [reply]
did you find any solution?? I have the same problem.

**Reply by spaceboi3000** [reply]
Make sure than in the file launch.json, you have this line "stopAtEntry": false. I had it true by accident so it caused the exact same error as you. I hope this helped.
